shared/middlewares/security.js

```diff
@@ -27,6 +28,71 @@ function securityMiddleware(server) {
   // does this by setting the X-Content-Type-Options header to nosniff.
   // @see https://helmetjs.github.io/docs/dont-sniff-mimetype/
   server.use(helmet.noSniff());
+
+  if (enableNonce) {
+    // Attach a unique "nonce" to every response. This allows use to declare
+    // inline scripts as being safe for execution against our content security policy.
+    // @see https://helmetjs.github.io/docs/csp/
+    server.use((request, response, next) => {
+      response.locals.nonce = uuid.v4();
+      next();
+    });
+  }
+
+  // Content Security Policy (CSP)
+  // It can be a pain to manage these, but it's a really great habit to get in to.
+  // @see https://helmetjs.github.io/docs/csp/
+  const cspConfig = {
+    directives: {
+      // The default-src is the default policy for loading content such as
+      // JavaScript, Images, CSS, Fonts, AJAX requests, Frames, HTML5 Media.
+      // As you might suspect, is used as fallback for unspecified directives.
+      defaultSrc: ["'self'"],
+
+      // Defines valid sources of JavaScript.
+      scriptSrc: [
+        "'self'",
+        'www.google-analytics.com',
+        'cdn.ravenjs.com',
+        'cdn.polyfill.io',
+
+        // Note: We will execution of any inline scripts that have the following
+        // nonce identifier attached to them.
+        // This is useful for guarding your application whilst allowing an inline
+        // script to do data store rehydration (redux/mobx/apollo) for example.
+        // @see https://helmetjs.github.io/docs/csp/
+        (request, response) => `'nonce-${response.locals.nonce}'`,
+      ],
+
+      // Defines the origins from which images can be loaded.
+      imgSrc: ["'self'", 'www.google-analytics.com', 'spectrum.imgix.net'],
+
+      // Defines valid sources of stylesheets.
+      styleSrc: ["'self'", "'unsafe-inline'"],
```

Comment:
`unsafe-inline` is here because we use styled-components, right?

If you set the `__webpack-nonce__` variable in `hyperion/renderer.js` styled-components should theoretically pick that up and add it to all its style tags, so we get rid of the `unsafe-inline` maybe?

Reply:
Oh that's awesome ! for sure we get rid of the unsafe-inline !!

Reply:
Hmm that's very weird. /cc @johannesnagl how did you make it work?

Reply:
Oh damn :confused: Thank you anyway, I guess let's just keep it with `unsafe-inline` for now.

Reply:
@mxstbr Ok Found it ! 
In fact it works great (not awake this morning 😴), set the `__webpack-nonce__` variable in `hyperion/renderer.js` apply correctly the nonce in all style tags.
But unfortunately we have also *inline styles* in the app and `styleSrc` directive includes `<style>` elements and also `style=` attributes.
We need to keep `unsafe-inline` for now. 😕